File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def getHeight(root, height):
            if not root:
                return height
            elif height < 0:
                return height
            else:
                left = getHeight(root.left, height + 1)
                right = getHeight(root.right, height + 1)
                if left < 0 or right < 0 or abs(left-right) > 1:
                    return -1
                else:
                    return max(left, right)
        if not root:
            return True
        height = getHeight(root, 0)
        if height < 0:
            return False
        else:
            return True
        

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        int height = checkBalance(root, 0);
        return height == -1 ? false : true;
    }   

    public int checkBalance(TreeNode node, int height){
        if(node == null){
            return 0;
        } else {
            int left = checkBalance(node.left, height + 1);
            int right = checkBalance(node.right, height - 1);
            if(left == -1 || right == -1){
                return -1;
            }
            if(Math.abs(left - right) <= 1){
                return Math.max(left, right) + 1;
            } else {
                return -1;
            }
        }
    }
}